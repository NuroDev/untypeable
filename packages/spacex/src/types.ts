export * from "./_shared/_shared.types";
export * from "./capsules/capsules.types";
export * from "./company/company.types";
export * from "./cores/cores.types";
export * from "./crew/crew.types";
export * from "./dragons/dragons.types";
export * from "./history/history.types";
export * from "./landpads/landpads.types";
export * from "./launches/launches.types";
export * from "./launchpads/launchpads.types";
export * from "./payloads/payloads.types";
export * from "./roadster/roadster.types";
export * from "./rockets/rockets.types";
export * from "./ships/ships.types";
export * from "./starlink/starlink.types";
