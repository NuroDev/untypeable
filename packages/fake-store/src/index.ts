import { initUntypeable } from "untypeable";

import type {
  Cart,
  CartParams,
  Carts,
  CartsParams,
  CreateCartParams,
  DeleteCartParams,
  UpdateCartParams,
  UserCartsParams,
} from "./cart/cart.types";
import type {
  Product,
  ProductsParams,
  Products,
  CreateProductParams,
  UpdateProductParams,
  ProductParams,
  ProductsCategories,
  ProductsCategoriesParams,
  ProductsCategoryParams,
  ProductsCategory,
  DeleteProductParams,
  UpdateProduct,
  CreateProduct,
} from "./products/products.types";

const u = initUntypeable().args<
  "GET" | "POST" | "PUT" | "PATCH" | "DELETE",
  string
>();

const cartRouter = u.router({
  DELETE: {
    /** Delete a cart */
    "/carts/:id": u.input<DeleteCartParams>().output<Cart>(),
  },
  GET: {
    /** Get all carts */
    "/carts": u.input<CartsParams>().output<Carts>(),
    /** Get a single cart */
    "/carts/:id": u.input<CartParams>().output<Cart>(),
    /** Get user carts */
    "/carts/user/:userId": u.input<UserCartsParams>().output<UserCartsParams>(),
  },
  PATCH: {
    /** Update a product */
    "/carts/:id": u.input<UpdateCartParams>().output<Cart>(),
  },
  POST: {
    /** Add a new product */
    "/carts": u.input<CreateCartParams>().output<Cart>(),
  },
  PUT: {
    /** Update a product */
    "/carts/:id": u.input<UpdateCartParams>().output<Cart>(),
  },
});

const loginRouter = u.router({
  DELETE: {},
  GET: {},
  PATCH: {},
  POST: {},
  PUT: {},
});

const productRouter = u.router({
  DELETE: {
    /** Delete a product */
    "/products/:id": u.input<DeleteProductParams>().output<Product>(),
  },
  GET: {
    /** Get all products */
    "/products": u.input<ProductsParams>().output<Products>(),
    /** Get a single product */
    "/products/:id": u.input<ProductParams>().output<Product>(),
    /** Get all categories */
    "/products/categories": u
      .input<ProductsCategoriesParams>()
      .output<ProductsCategories>(),
    /** Get products in a specific category */
    "/products/category/:category": u
      .input<ProductsCategoryParams>()
      .output<ProductsCategory>(),
  },
  PATCH: {
    /** Update a product */
    "/products/:id": u.input<UpdateProductParams>().output<UpdateProduct>(),
  },
  POST: {
    /** Add a new product */
    "/products": u.input<CreateProductParams>().output<CreateProduct>(),
  },
  PUT: {
    /** Update a product */
    "/products/:id": u.input<UpdateProductParams>().output<UpdateProduct>(),
  },
});

const userRouter = u.router({
  DELETE: {},
  GET: {},
  PATCH: {},
  POST: {},
  PUT: {},
});

const router = u
  .router({})
  .merge(productRouter)
  .merge(cartRouter)
  .merge(userRouter)
  .merge(loginRouter);

export type FakeStoreRouter = typeof router;
