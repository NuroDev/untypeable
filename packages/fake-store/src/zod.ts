export * from "./cart/cart.validators";
export * from "./products/products.validators";
