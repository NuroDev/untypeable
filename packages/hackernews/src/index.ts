import { initUntypeable } from "untypeable";

import type { Item, ItemParams } from "./items/items.types";

const u = initUntypeable();

const router = u.router({
  "/v0/item/:id.json": u.input<ItemParams>().output<Item>(),
});

export type HackerNewsRouter = typeof router;

export * from "./items/items.types";
