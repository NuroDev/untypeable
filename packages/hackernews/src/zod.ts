export * from "./items/items.validators";
