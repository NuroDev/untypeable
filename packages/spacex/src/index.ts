import { initUntypeable } from "untypeable";

import type {
  Capsule,
  CapsuleParams,
  Capsules,
} from "./capsules/capsules.types";

const u = initUntypeable();

const router = u.router({
  "/capsules": u.output<Capsules>(),
  "/capsules/:id": u.input<CapsuleParams>().output<Capsule>(),
});

export type SpaceXRouter = typeof router;
