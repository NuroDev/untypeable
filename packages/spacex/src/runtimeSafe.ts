import { initUntypeable } from "untypeable";

import {
  CapsuleSchema,
  CapsuleParamsSchema,
  CapsulesSchema,
} from "./capsules/capsules.validators";
import { CompanySchema } from "./company/company.validators";
import {
  CoreSchema,
  CoreParamsSchema,
  CoresSchema,
} from "./cores/cores.validators";
import {
  CrewSchema,
  CrewMemberSchema,
  CrewMemberParamsSchema,
} from "./crew/crew.validators";
import {
  DragonSchema,
  DragonParamsSchema,
  DragonsSchema,
} from "./dragons/dragons.validators";
import {
  HistoryEventSchema,
  HistoryEventParamsSchema,
  HistoryEventsSchema,
} from "./history/history.validators";
import {
  LandPadSchema,
  LandPadParamsSchema,
  LandPadsSchema,
} from "./landpads/landpads.validators";
import {
  LaunchesV4Schema,
  LaunchesV5Schema,
  LaunchParamsSchema,
  LaunchV4Schema,
  LaunchV5Schema,
} from "./launches/launches.validators";
import {
  LaunchPadSchema,
  LaunchPadParamsSchema,
  LaunchPadsSchema,
} from "./launchpads/launchpads.validators";
import {
  PayloadSchema,
  PayloadParamsSchema,
  PayloadsSchema,
} from "./payloads/payloads.validators";
import { RoadsterSchema } from "./roadster/roadster.validators";
import {
  RocketSchema,
  RocketParamsSchema,
  RocketsSchema,
} from "./rockets/rockets.validators";
import {
  ShipSchema,
  ShipParamsSchema,
  ShipsSchema,
} from "./ships/ships.validators";
import {
  StarlinkSatelliteSchema,
  StarlinkSatelliteParamsSchema,
  StarlinkSatellitesSchema,
} from "./starlink/starlink.validators";

const u = initUntypeable();

export const spaceXSafeRouter = u.router({
  "/v4/capsules": u.output(CapsulesSchema),
  "/v4/capsules/:id": u.input(CapsuleParamsSchema).output(CapsuleSchema),
  "/v4/company": u.output(CompanySchema),
  "/v4/cores": u.output(CoresSchema),
  "/v4/cores/:id": u.input(CoreParamsSchema).output(CoreSchema),
  "/v4/crew": u.output(CrewSchema),
  "/v4/crew/:id": u.input(CrewMemberParamsSchema).output(CrewMemberSchema),
  "/v4/dragons": u.output(DragonsSchema),
  "/v4/dragons/:id": u.input(DragonParamsSchema).output(DragonSchema),
  "/v4/history": u.output(HistoryEventsSchema),
  "/v4/history/:id": u
    .input(HistoryEventParamsSchema)
    .output(HistoryEventSchema),
  "/v4/landpads": u.output(LandPadsSchema),
  "/v4/landpads/:id": u.input(LandPadParamsSchema).output(LandPadSchema),
  "/v4/launches": u.output(LaunchesV4Schema),
  "/v4/launches/latest": u.output(LaunchV4Schema),
  "/v4/launches/next": u.output(LaunchV4Schema),
  "/v4/launches/past": u.output(LaunchesV4Schema),
  "/v4/launches/upcoming": u.output(LaunchesV4Schema),
  "/v4/launches/:id": u.input(LaunchParamsSchema).output(LaunchV4Schema),
  "/v5/launches": u.output(LaunchesV5Schema),
  "/v5/launches/latest": u.output(LaunchV5Schema),
  "/v5/launches/next": u.output(LaunchV5Schema),
  "/v5/launches/past": u.output(LaunchesV5Schema),
  "/v5/launches/upcoming": u.output(LaunchesV5Schema),
  "/v5/launches/:id": u.input(LaunchParamsSchema).output(LaunchV5Schema),
  "/v4/launchpads": u.output(LaunchPadsSchema),
  "/v4/launchpads/:id": u.input(LaunchPadParamsSchema).output(LaunchPadSchema),
  "/v4/payloads": u.output(PayloadsSchema),
  "/v4/payloads/:id": u.input(PayloadParamsSchema).output(PayloadSchema),
  "/v4/roadster": u.output(RoadsterSchema),
  "/v4/rockets": u.output(RocketsSchema),
  "/v4/rockets/:id": u.input(RocketParamsSchema).output(RocketSchema),
  "/v4/ships": u.output(ShipsSchema),
  "/v4/ships/:id": u.input(ShipParamsSchema).output(ShipSchema),
  "/v4/starlink": u.output(StarlinkSatellitesSchema),
  "/v4/starlink/:id": u
    .input(StarlinkSatelliteParamsSchema)
    .output(StarlinkSatelliteSchema),
});
