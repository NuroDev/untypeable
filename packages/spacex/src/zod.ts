export * from "./capsules/capsules.validators";
export * from "./company/company.validators";
export * from "./cores/cores.validators";
export * from "./crew/crew.validators";
export * from "./dragons/dragons.validators";
export * from "./landpads/landpads.validators";
export * from "./starlink/starlink.validators";
