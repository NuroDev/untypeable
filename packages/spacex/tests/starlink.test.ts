import { describe, it } from "vitest";

import { StarlinkSatelliteSchema, StarlinkSatellitesSchema } from "../src/zod";
import { useTestClient } from "./_client";

describe.concurrent("SpaceX - Starlink", () => {
  const client = useTestClient();

  it("GET - /v4/starlink", async ({ expect }) => {
    const starlinkSatellites = await client("/v4/starlink");

    expect(starlinkSatellites).toBeDefined();
    expect(Array.isArray(starlinkSatellites)).toBe(true);
    expect(starlinkSatellites.at(0)).toBeDefined();
    expect(starlinkSatellites.at(0)).toMatchSnapshot({
      height_km: null,
      id: "5eed770f096e59000698560d",
      latitude: null,
      launch: "5eb87d30ffd86e000604b378",
      longitude: null,
      spaceTrack: {
        CCSDS_OMM_VERS: "2.0",
        COMMENT: "GENERATED VIA SPACE-TRACK.ORG API",
        CREATION_DATE: "2020-10-13T04:16:08",
        ORIGINATOR: "18 SPCS",
        OBJECT_NAME: "STARLINK-30",
        OBJECT_ID: "2019-029K",
        CENTER_NAME: "EARTH",
        REF_FRAME: "TEME",
        TIME_SYSTEM: "UTC",
        MEAN_ELEMENT_THEORY: "SGP4",
        EPOCH: "2020-10-13T02:56:59.566560",
        MEAN_MOTION: 16.43170483,
        ECCENTRICITY: 0.0003711,
        INCLINATION: 52.9708,
        RA_OF_ASC_NODE: 332.0356,
        ARG_OF_PERICENTER: 120.7278,
        MEAN_ANOMALY: 242.0157,
        EPHEMERIS_TYPE: 0,
        CLASSIFICATION_TYPE: "U",
        NORAD_CAT_ID: 44244,
        ELEMENT_SET_NO: 999,
        REV_AT_EPOCH: 7775,
        BSTAR: 0.0022139,
        MEAN_MOTION_DOT: 0.47180237,
        MEAN_MOTION_DDOT: 0.000012426,
        SEMIMAJOR_AXIS: 6535.519,
        PERIOD: 87.635,
        APOAPSIS: 159.809,
        PERIAPSIS: 154.958,
        OBJECT_TYPE: "PAYLOAD",
        RCS_SIZE: "LARGE",
        COUNTRY_CODE: "US",
        LAUNCH_DATE: "2019-05-24",
        SITE: "AFETR",
        DECAY_DATE: "2020-10-13",
        DECAYED: 1,
        FILE: 2850561,
        GP_ID: 163365918,
        TLE_LINE0: "0 STARLINK-30",
        TLE_LINE1:
          "1 44244U 19029K   20287.12291165  .47180237  12426-4  22139-2 0  9995",
        TLE_LINE2:
          "2 44244  52.9708 332.0356 0003711 120.7278 242.0157 16.43170483 77756",
      },
      velocity_kms: null,
      version: "v0.9",
    });

    expect(StarlinkSatellitesSchema.safeParse(starlinkSatellites).success).toBe(
      true
    );
  });

  it("GET - /v4/starlink/:id", async ({ expect }) => {
    const starlinkSatellite = await client("/v4/starlink/:id", {
      id: "5eed770f096e59000698560d",
    });

    expect(starlinkSatellite).toBeDefined();
    expect(starlinkSatellite).toMatchSnapshot({
      height_km: null,
      id: "5eed770f096e59000698560d",
      latitude: null,
      launch: "5eb87d30ffd86e000604b378",
      longitude: null,
      spaceTrack: {
        CCSDS_OMM_VERS: "2.0",
        COMMENT: "GENERATED VIA SPACE-TRACK.ORG API",
        CREATION_DATE: "2020-10-13T04:16:08",
        ORIGINATOR: "18 SPCS",
        OBJECT_NAME: "STARLINK-30",
        OBJECT_ID: "2019-029K",
        CENTER_NAME: "EARTH",
        REF_FRAME: "TEME",
        TIME_SYSTEM: "UTC",
        MEAN_ELEMENT_THEORY: "SGP4",
        EPOCH: "2020-10-13T02:56:59.566560",
        MEAN_MOTION: 16.43170483,
        ECCENTRICITY: 0.0003711,
        INCLINATION: 52.9708,
        RA_OF_ASC_NODE: 332.0356,
        ARG_OF_PERICENTER: 120.7278,
        MEAN_ANOMALY: 242.0157,
        EPHEMERIS_TYPE: 0,
        CLASSIFICATION_TYPE: "U",
        NORAD_CAT_ID: 44244,
        ELEMENT_SET_NO: 999,
        REV_AT_EPOCH: 7775,
        BSTAR: 0.0022139,
        MEAN_MOTION_DOT: 0.47180237,
        MEAN_MOTION_DDOT: 0.000012426,
        SEMIMAJOR_AXIS: 6535.519,
        PERIOD: 87.635,
        APOAPSIS: 159.809,
        PERIAPSIS: 154.958,
        OBJECT_TYPE: "PAYLOAD",
        RCS_SIZE: "LARGE",
        COUNTRY_CODE: "US",
        LAUNCH_DATE: "2019-05-24",
        SITE: "AFETR",
        DECAY_DATE: "2020-10-13",
        DECAYED: 1,
        FILE: 2850561,
        GP_ID: 163365918,
        TLE_LINE0: "0 STARLINK-30",
        TLE_LINE1:
          "1 44244U 19029K   20287.12291165  .47180237  12426-4  22139-2 0  9995",
        TLE_LINE2:
          "2 44244  52.9708 332.0356 0003711 120.7278 242.0157 16.43170483 77756",
      },
      velocity_kms: null,
      version: "v0.9",
    });

    expect(StarlinkSatelliteSchema.safeParse(starlinkSatellite).success).toBe(
      true
    );
  });
});
