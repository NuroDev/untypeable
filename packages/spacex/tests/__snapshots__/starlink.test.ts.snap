// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SpaceX - Starlink > GET - /v4/starlink 1`] = `
{
  "height_km": null,
  "id": "5eed770f096e59000698560d",
  "latitude": null,
  "launch": "5eb87d30ffd86e000604b378",
  "longitude": null,
  "spaceTrack": {
    "APOAPSIS": 159.809,
    "ARG_OF_PERICENTER": 120.7278,
    "BSTAR": 0.0022139,
    "CCSDS_OMM_VERS": "2.0",
    "CENTER_NAME": "EARTH",
    "CLASSIFICATION_TYPE": "U",
    "COMMENT": "GENERATED VIA SPACE-TRACK.ORG API",
    "COUNTRY_CODE": "US",
    "CREATION_DATE": "2020-10-13T04:16:08",
    "DECAYED": 1,
    "DECAY_DATE": "2020-10-13",
    "ECCENTRICITY": 0.0003711,
    "ELEMENT_SET_NO": 999,
    "EPHEMERIS_TYPE": 0,
    "EPOCH": "2020-10-13T02:56:59.566560",
    "FILE": 2850561,
    "GP_ID": 163365918,
    "INCLINATION": 52.9708,
    "LAUNCH_DATE": "2019-05-24",
    "MEAN_ANOMALY": 242.0157,
    "MEAN_ELEMENT_THEORY": "SGP4",
    "MEAN_MOTION": 16.43170483,
    "MEAN_MOTION_DDOT": 0.000012426,
    "MEAN_MOTION_DOT": 0.47180237,
    "NORAD_CAT_ID": 44244,
    "OBJECT_ID": "2019-029K",
    "OBJECT_NAME": "STARLINK-30",
    "OBJECT_TYPE": "PAYLOAD",
    "ORIGINATOR": "18 SPCS",
    "PERIAPSIS": 154.958,
    "PERIOD": 87.635,
    "RA_OF_ASC_NODE": 332.0356,
    "RCS_SIZE": "LARGE",
    "REF_FRAME": "TEME",
    "REV_AT_EPOCH": 7775,
    "SEMIMAJOR_AXIS": 6535.519,
    "SITE": "AFETR",
    "TIME_SYSTEM": "UTC",
    "TLE_LINE0": "0 STARLINK-30",
    "TLE_LINE1": "1 44244U 19029K   20287.12291165  .47180237  12426-4  22139-2 0  9995",
    "TLE_LINE2": "2 44244  52.9708 332.0356 0003711 120.7278 242.0157 16.43170483 77756",
  },
  "velocity_kms": null,
  "version": "v0.9",
}
`;

exports[`SpaceX - Starlink > GET - /v4/starlink/:id 1`] = `
{
  "height_km": null,
  "id": "5eed770f096e59000698560d",
  "latitude": null,
  "launch": "5eb87d30ffd86e000604b378",
  "longitude": null,
  "spaceTrack": {
    "APOAPSIS": 159.809,
    "ARG_OF_PERICENTER": 120.7278,
    "BSTAR": 0.0022139,
    "CCSDS_OMM_VERS": "2.0",
    "CENTER_NAME": "EARTH",
    "CLASSIFICATION_TYPE": "U",
    "COMMENT": "GENERATED VIA SPACE-TRACK.ORG API",
    "COUNTRY_CODE": "US",
    "CREATION_DATE": "2020-10-13T04:16:08",
    "DECAYED": 1,
    "DECAY_DATE": "2020-10-13",
    "ECCENTRICITY": 0.0003711,
    "ELEMENT_SET_NO": 999,
    "EPHEMERIS_TYPE": 0,
    "EPOCH": "2020-10-13T02:56:59.566560",
    "FILE": 2850561,
    "GP_ID": 163365918,
    "INCLINATION": 52.9708,
    "LAUNCH_DATE": "2019-05-24",
    "MEAN_ANOMALY": 242.0157,
    "MEAN_ELEMENT_THEORY": "SGP4",
    "MEAN_MOTION": 16.43170483,
    "MEAN_MOTION_DDOT": 0.000012426,
    "MEAN_MOTION_DOT": 0.47180237,
    "NORAD_CAT_ID": 44244,
    "OBJECT_ID": "2019-029K",
    "OBJECT_NAME": "STARLINK-30",
    "OBJECT_TYPE": "PAYLOAD",
    "ORIGINATOR": "18 SPCS",
    "PERIAPSIS": 154.958,
    "PERIOD": 87.635,
    "RA_OF_ASC_NODE": 332.0356,
    "RCS_SIZE": "LARGE",
    "REF_FRAME": "TEME",
    "REV_AT_EPOCH": 7775,
    "SEMIMAJOR_AXIS": 6535.519,
    "SITE": "AFETR",
    "TIME_SYSTEM": "UTC",
    "TLE_LINE0": "0 STARLINK-30",
    "TLE_LINE1": "1 44244U 19029K   20287.12291165  .47180237  12426-4  22139-2 0  9995",
    "TLE_LINE2": "2 44244  52.9708 332.0356 0003711 120.7278 242.0157 16.43170483 77756",
  },
  "velocity_kms": null,
  "version": "v0.9",
}
`;
